crates/chain/tests/test_tx_graph.rs

```diff
@@ -1325,3 +1330,89 @@ fn tx_graph_update_conversion() {
         );
     }
 }
+
+#[test]
+fn test_seen_at_updates() {
+    // Update both first_seen and last_seen
+    let seen_at = 1000000_u64;
+    let mut graph = TxGraph::<BlockId>::default();
+    let mut changeset = graph.insert_seen_at(hash!("tx1"), seen_at);
+    assert_eq!(
+        changeset,
+        ChangeSet {
+            txs: [].into(),
+            txouts: [].into(),
+            anchors: [].into(),
+            first_seen: [(hash!("tx1"), 1000000)].into(),
+            last_seen: [(hash!("tx1"), 1000000)].into(),
+            last_evicted: [].into(),
+        }
+    );
+    graph.apply_changeset(changeset);
```

Comment:
I don't think this is needed. `graph.insert_seen_at` should already mutate the internal state, so this line essentially would do nothing if everything is working correctly.